src/test/run-pass/issue-21058.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2014 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![allow(unstable)]
+
+struct NT(str);
+struct DST { a: u32, b: str }
+
+fn main() {
+    // get_tydesc should support unsized types
+    unsafe {
+        // Slice
+        std::intrinsics::get_tydesc::<[u8]>();
```

Comment:
I don't think it's defined what the names are supposed to be, so I'm not sure how useful that would be. On the other hand it does at least ensure that they're not being optimised out or anything stupid, so I'll add that.